stdlib/fun.mli

```diff
@@ -61,3 +63,81 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+(** {1:examples Examples}
+
+    {2 Combinators}
+
+    {{!combinators}Combinators} provide a lightweight and sometimes more
+    readable way to create anonymous functions, best used as short-lived
+    arguments rather than standalone definitions. The examples below will
+    demonstrate this mainly with the {!module:List} module.
+
+    {!val:id}
+    {[
+    # List.init 3 Fun.id;;
+    - : int list = [0; 1; 2]
+
+    # List.filter_map Fun.id [None; Some 2; Some 3; None; Some 5];;
+    - : int list = [2; 3; 5]
+
+    # let to_flat = Float.Array.map_from_array Fun.id
+    val to_flat : float array -> Float.Array.t
+    ]}
```

Comment:
I think it might make sense to add an one sentence description to each examples.

Reply:
Added descriptions